loader/assets/annotate.py

```diff
@@ -46,15 +46,38 @@ def nameOf(element):
 
         return tids[tid]
     
+    ui.log.log("  annotate Element...")
     ElementRoot = haven.find("Element")
-    # Annotate Elements
+    ElementName = {}
+    ElementLink = {}
+    # Annotate Elements and create list of links.
     for element in ElementRoot:
         mid = element.get("mid")
-
         objectInfo = element.find("objectInfo")
         if objectInfo is not None:
             element.set("_annotation", nameOf(objectInfo))
-    
+            ElementName[mid] = element.get("_annotation")
+        # Keep track of links, inverted.
+        linked = element.find("linked")
+        for link in linked.findall("l"):
+            linkid = link.get("id")
+            if linkid is not None and linkid not in ElementLink:
+                ElementLink[linkid] = []
+            if mid not in ElementLink[linkid]:
+                ElementLink[linkid].append(mid)
+
+    # Element Second pass, give linked-by reference list.
+    for element in ElementRoot:
+        mid = element.get("mid")
+        if mid in ElementLink:
+            s=""
+            for link in ElementLink[mid]:
+                s = s + link
+                if link in ElementName:
+                    s = s + " " + ElementName[link]
+                s = s + ";"
```

Comment:
Let's do this via string.join() instead. List comprehension will also make the semantic flow a little bit more clear.
```suggestion
            links = (
                (link + " " + ElementName[link]) if link in ElementName else link
                for link in ElementLink[mid]
            )
            s = "; ".join(links)
```
Result:
```xml
<me _linkedBy="1871 CO2 Producer; 1908 Assembler; 922 Gas scrubber; 861; 862; 924" >
```

Reply:
Sounds good.
I was also thinking of using a dash after the mid in links, instead of a space, to make it easier to parse by other tools.

Eventually this will make many passes until every Element will have a meaningful annotation if at all possible, then links can give more of a hint.
Before that, I need to fix the texture and animations annotations.
Then I can move on to reports.

Reply:
Because of the way pull requests work, you'll need to make the change on your branch. Unresolving this until then.
Thankfully the suggestion feature means you should be able to just click the "add suggestion to branch" button.

Reply:
Correction. It looks like you have to use "commit suggestion". This isn't a feature I'm familiar with yet, unfortunately; Github seems to be implying I could force commit that to your branch, but that seems... unlikely.

Reply:
Fixed and pushed to my branch.